### ISSUE TITLE
`reportIncompatibleUnannotatedOverride` false positive when overridden type has an incompatible inferred type

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1247]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.13&typeCheckingMode=all&reportUnannotatedClassAttribute=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2AsAFB10DGl5Azq1AGJhgBcdUQVHK9CxMlQDaARgA0UAEwBdKAF4o0hrWZsOAIXIgAFNzABKfrSFQAJiWBQA%2Bo9TFnR1iUrALA60M9vADpyNUUoAGIoMnAQUQB5ADcYpDtYRBIoACJUGCzMDhQwfBgAC0zWcghMtigAIzZM%2BARMrKJSCkoZeWUsrToQEmSqR2aSIwNjMxCzSLEOqTlFJSA)

```python
from typing import Literal


class Foo:
    a: Literal[1, 2] = 1

class Bar(Foo):
    def __init__(self):
        self.a = 2 # error: Override type "int" is not the same as base type "Literal[1, 2]"

# this is safe because the type doesn't actually get widened to int
reveal_type(Bar().a) # Literal[1, 2]
```

**Comment by DetachHead** [comment]
another example from #1269:

```py
class Animal:
    pass

class Snake(Animal):
    pass

class Cat(Animal):
    pass

class Foo:
    A: Animal = Snake()  # No issue, Snake is an Animal

class Bar(Foo):
    A = Cat()  #  "A" overrides symbol of same name in class "Foo"
               #    Variable is mutable so its type is invariant
               #       Override type "Cat" is not the same as base type "Animal"  (reportIncompatibleUnannotatedOverride)
```
